apps/editing-toolkit/editing-toolkit-plugin/premium-content/blocks/container/edit.js

```diff
@@ -330,41 +339,77 @@ function Edit( props ) {
 		);
 	}
 
-	let stripeNudge = null;
-
-	if ( ! shouldUpgrade && apiState !== API_STATE_CONNECTED && connectURL ) {
-		const stripeConnectUrl = getConnectUrl( props, connectURL );
+	const shouldShowConnectButton = () => {
+		if ( ! shouldUpgrade && apiState !== API_STATE_CONNECTED && connectURL ) {
+			return true;
+		}
 
-		stripeNudge = <StripeNudge { ...props } stripeConnectUrl={ stripeConnectUrl } />;
-	}
+		return false;
+	};
 
 	return (
-		<div className={ className } ref={ wrapperRef }>
-			{ props.noticeUI }
-			{ ( isSelected || selectedInnerBlock ) && apiState === API_STATE_CONNECTED && (
-				<Controls
-					{ ...props }
-					plans={ products }
-					selectedPlanId={ props.attributes.selectedPlanId }
-					onSelected={ selectPlan }
-					getPlanDescription={ getPlanDescription }
-				/>
-			) }
-			{ ( isSelected || selectedInnerBlock ) && apiState === API_STATE_CONNECTED && (
-				<Inspector { ...props } savePlan={ savePlan } siteSlug={ siteSlug } />
-			) }
-			{ ( isSelected || selectedInnerBlock ) && (
-				<Tabs { ...props } tabs={ tabs } selectedTab={ selectedTab } onSelected={ selectTab } />
-			) }
-			<Context.Provider
-				value={ {
-					selectedTab,
-					stripeNudge,
-				} }
-			>
-				<Blocks />
-			</Context.Provider>
-		</div>
+		<>
+			<BlockControls>
+				{ shouldShowConnectButton() && (
+					<ToolbarGroup>
+						<ToolbarButton
+							icon={ flashIcon }
+							onClick={ ( e ) => {
+								props.autosaveAndRedirect( e, getConnectUrl( props, connectURL ) );
+							} }
+							className="connect-stripe components-tab-button"
+						>
+							{ __( 'Connect Stripe', 'full-site-editing' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **28** times:
`translate( 'Stripe Connection' )` *ES Score: 6*